modules/tensor_mechanics/src/kernels/InertialForce.C

```diff
@@ -42,33 +44,34 @@ InertialForce::validParams()
   return params;
 }
 
-InertialForce::InertialForce(const InputParameters & parameters)
-  : TimeKernel(parameters),
-    _density(getMaterialProperty<Real>("density")),
-    _has_beta(isParamValid("beta")),
-    _has_gamma(isParamValid("gamma")),
-    _beta(_has_beta ? getParam<Real>("beta") : 0.1),
-    _gamma(_has_gamma ? getParam<Real>("gamma") : 0.1),
-    _has_velocity(isParamValid("velocity")),
-    _has_acceleration(isParamValid("acceleration")),
-    _eta(getMaterialProperty<Real>("eta")),
-    _alpha(getParam<Real>("alpha")),
+template <bool is_ad>
+InertialForceTempl<is_ad>::InertialForceTempl(const InputParameters & parameters)
+  : InertialForceParent<is_ad>(parameters),
+    _density(this->template getGenericMaterialProperty<Real, is_ad>("density")),
+    _has_beta(this->isParamValid("beta")),
+    _has_gamma(this->isParamValid("gamma")),
+    _beta(_has_beta ? this->template getParam<Real>("beta") : 0.1),
+    _gamma(_has_gamma ? this->template getParam<Real>("gamma") : 0.1),
+    _has_velocity(this->isParamValid("velocity")),
+    _has_acceleration(this->isParamValid("acceleration")),
+    _eta(this->template getGenericMaterialProperty<Real, is_ad>("eta")),
+    _alpha(this->template getParam<Real>("alpha")),
     _time_integrator(*_sys.getTimeIntegrator())
 {
   if (_has_beta && _has_gamma && _has_velocity && _has_acceleration)
   {
-    _vel_old = &coupledValueOld("velocity");
-    _accel_old = &coupledValueOld("acceleration");
-    _u_old = &valueOld();
+    _vel_old = &this->coupledValueOld("velocity");
+    _accel_old = &this->coupledValueOld("acceleration");
+    _u_old = &this->valueOld();
```

Comment:
we dont use this_> to class member functions much in moose. 
why did you introduce those?

Reply:
I was wondering the same thing but didn't ask. I assumed  it was something required by the templating for AD/non-AD. Is that the case? If it's not required, I'd get rid of it.

Reply:
Yes, it is due to the new templating

Reply:
so what do we prefer:
have a Coupleable::coupledValueOld in the header
OR
using this->

Reply:
just need an answer here then green button